src/main/java/com/aws/greengrass/componentmanager/KernelConfigResolver.java

```diff
@@ -112,20 +114,23 @@ public KernelConfigResolver(ComponentStore componentStore, Kernel kernel, Nucleu
         this.deviceConfiguration = deviceConfiguration;
 
         // More system parameters can be added over time by extending this map with new namespaces/keys
-        HashMap<String, CrashableFunction<ComponentIdentifier, String, IOException>> artifactNamespace =
-                new HashMap<>();
+        Map<String, CrashableFunction<ComponentIdentifier, String, IOException>> artifactNamespace = new HashMap<>();
```

Comment:
nit: `HashMap` shorthand might help here:
```
systemParameters.put(ARTIFACTS_NAMESPACE, new HashMap<String, CrashableFunction<ComponentIdentifier, String, IOException>>() {{
    put(PATH_KEY, (id) -> nucleusPaths.artifactPath(id).toAbsolutePath().toString());
    put(DECOMPRESSED_PATH_KEY, (id) -> nucleusPaths.unarchiveArtifactPath(id).toAbsolutePath().toString());
}});
```
etc. May prevent some namespace -> value setting mistakes for system parameters.